### ISSUE TITLE
Expression 'undefined' in attribute 'dataurl' error

### ISSUE THREAD

**Issue by sgentile** [legalthings/angular-signature#14]
angular.js:13550 Error: [$compile:nonassign] Expression 'undefined' in attribute 'dataurl' used with directive 'signaturePad' is non-assignable!

my use case is fairly simple - almost straight from the sample

```
<div>
    <div class="modal-header">
        <button type="button" class="close pull-right" ng-click="$dismiss()" aria-hidden="true">&times;</button>
        <h4 class="modal-title">{{vm.label}}</h4>
    </div>
    <div class="modal-body">
        <p>
            Sign here:
            <div class="sig-border">
                <signature-pad accept="vm.accept" clear="clear"></signature-pad>
            </div>
        </p>
    </div>
    <div class="modal-footer">
        <button type="button" class="btn btn-default" ng-click="$dismiss()">Cancel</button>
        <button type="button" class="btn btn-default" ng-click="clear()">Clear</button>
        <button type="button" class="btn btn-primary" ng-click="vm.sign()">Sign</button>
    </div>
</div>
```

```
vm.sign = function(){
            var signature = vm.accept();
            if (!signature.isEmpty) {
                $scope.$close(signature.dataUrl);
            }
        };
```

**Comment by vincentbitter** [comment]
Had the same issue, but if you add a dataurl="" attribute to the <signature-pad> element, it'll be gone.

**Comment by firesnatch** [comment]
Thanks @vincentbitter, that works.  The documentation should be updated to include this info.

**Comment by rbwinslow** [comment]
I still get an error, though slightly different, after following @vincentbitter 's advice:

`Expression '' in attribute 'dataurl' used with directive 'signaturePad' is non-assignable!`

I'm using the current head commit of the software (0853ce3a7242a810c00dbc10de33b217f9c8c7bb) and embedding the signature pad in a UI Bootstrap Tab, which requires the following shim (in my controller) to properly size the canvas:

```
$timeout(() => {
   $window.dispatchEvent(new Event('resize'));
});
```

**Comment by rbwinslow** [comment]
OK, my mistake. I wasn't assigning the "accept" and "clear" attributes to any variables, which is required as one can clearly see examining the code (and the fact that it's done in all of the examples in the doc). Sorry!

**Comment by jasny** [comment]
These attributes should be made optional

**Comment by jasny** [comment]
Fixed in #36
